File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem13/Problem13.java
public class Problem13 {
    
    public static void printHanoi(int n, char S, char D, char A){
        if (n == 1) {
            System.out.println(S + " -> " + D);
        } else {
            printHanoi(n - 1, S, A, D);
            System.out.println(S + " -> " + D);
            printHanoi(n - 1, A, D, S);
        }
    }

    public static void main(String[] args) {
        int n = 3;
        char S = 'S';
        char D = 'D';
        char A = 'A';

        printHanoi(n, S, D, A);
    }
}

File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_13/problem_13.py
def printHanoi(n, S, D, A):
    if n == 1:
        print(f"{S} -> {D}")
    else:
        printHanoi(n - 1, S, A, D)
        print(f"{S} -> {D}")
        printHanoi(n - 1, A, D, S)

def main():
    n = 3
    charS = 'S'
    charD = 'D'
    charA = 'A'
    
    printHanoi(n, charS, charD, charA)


if __name__ == "__main__":
    main()